### ISSUE TITLE
hidden navigation menu still visible to assistive devices on mobile

### ISSUE THREAD

**Issue by madalynrose** [gatsbyjs/gatsby#24625]
## Description

Screen reader reads hidden navigation menu items that the user cannot interact with on mobile.

### Steps to reproduce

1. Load up [www.gatsbyjs.org/](www.gatsbyjs.org/) on an iOS device using VoiceOver.
1. In the bottom bar, touch, then double-tap Tutorials.
1. Touch the "Secondary Navigation" button near the bottom right corner, right above the sticky bottom navigation bar. It is spoken as "collapsed".
1. Now, swipe to the right. 

### Expected result

You will hear all kinds of secondary navigation items that are actually not visible on the screen. Also, if trying to activate these, it will fail because VoiceOver will simulate a tap where it expects the item to be. Since it is off-screen, the tap goes nowhere.

### Actual result

VoiceOver focus goes to the bottom navigation bar instead, which is the next item visible on the screen.

### Environment

iOS using VoiceOver

### Possible Fix

This could be because the menu isn't hidden by display:none; visibility:hidden; or such, but only somehow moved to the background or off the view port. That, however, doesn't make it invisible to screen readers. If solving this via CSS isn't possible, aria-hidden="true" should be applied to the container of that menu when it is hidden, and the attribute should be removed once it is made visible.

**Comment by ehrencrona** [comment]
<img width="1100" alt="Screen Shot 2020-06-24 at 9 37 38 AM" src="https://user-images.githubusercontent.com/1862212/85557683-617b1b80-b5fe-11ea-851a-02c0b11fb327.png">

Seems like the navigation container is still on screen, just empty. 

The child div (with class `css-eie263` in this screenshot) has a `transform: translateX(-320px);` that moves it beyond the edge of the screen. I'd suspect that if you move it up to its parent div (the one selected in the screenshot) and possible change `transform` to `left`, the issue would be resolved.

I'm happy to try to fix it if someone can point me to the repo where the gatsby website is located. Or is it private?

**Comment by Pyrax** [comment]
@ehrencrona you can find the site in this same repo under "www". This might also help: [sticky sidebar component](https://github.com/gatsbyjs/gatsby/blob/master/www/src/components/sidebar/sticky-responsive-sidebar.js)

**Comment by ehrencrona** [comment]
I could reproduce the offscreen menu still being read by VoiceOver but I couldn't reproduce the menu items not being selectable. I could select menu items while offscreen and the page would update correctly. But anyway, I added `aria-hidden to fix it. Actually removing it from the DOM or using `display: none` would mess with the animations.
